src/Storages/StorageURL.cpp

```diff
@@ -858,6 +863,54 @@ FormatSettings StorageURL::getFormatSettingsFromArgs(const StorageFactory::Argum
     return format_settings;
 }
 
+ASTs::iterator StorageURL::collectHeaders(
+    ASTs & url_function_args, URLBasedDataSourceConfiguration & configuration, ContextPtr context)
+{
+    ASTs::iterator headers_it;
+
+    for (auto arg_it = url_function_args.begin(); arg_it != url_function_args.end(); ++arg_it)
+    {
+        const auto * headers_ast_function = (*arg_it)->as<ASTFunction>();
+        if (headers_ast_function && headers_ast_function->name == "headers")
+        {
+            if (headers_it != ASTs::iterator())
```

Comment:
It works but I'm not sure that it's a good way to check invalid iterator. I think it would be better to assign `headers_it` to `url_function_args.end()` and compare it with it here and after calling `collectHeaders` (it will look better, like if there is no headers in args we will return `url_function_args.end()`